lib/src/generator/html_generator.dart

```diff
@@ -40,6 +40,8 @@ class HtmlGeneratorBackend extends GeneratorBackendBase {
     var data = PackageTemplateData(options, graph, package);
     var content = templates.renderError(data);
     write(writer, '__404error.html', data, content);
+    var searchContent = templates.renderSearchPage(data);
+    write(writer, 'search_results_page.html', data, searchContent);
```

Comment:
The word 'page' is rather unnecessary here. Let's name this file something shorter like `search_results` or even just `search`? Maybe we want to reserve `search.html` for something else but I'm inclined to be as concise as possible, and if we add some big fancy search page later, then that and this search results page can both just be `search.html`.